owncloudApp/src/main/java/com/owncloud/android/presentation/ui/sharing/ShareActivity.kt

```diff
@@ -0,0 +1,385 @@
+            shareType,
+            shareeName,
+            getAppropiatePermissions(shareType)
+        )
+    }
+
+    private fun observePrivateShareCreation() {
+        ocShareViewModel.privateShareCreationStatus.observe(
+            this,
+            Observer { uiResult ->
+                when (uiResult?.status) {
+                    Status.ERROR -> {
+                        Snackbar.make(
+                            findViewById(android.R.id.content)!!,
+                            uiResult.errorMessage!!,
+                            Snackbar.LENGTH_SHORT
+                        ).show()
+                        dismissLoadingDialog()
+                    }
+                    Status.LOADING -> {
+                        showLoadingDialog(R.string.common_loading)
+                    }
+                    else -> {
+                        Log.d(
+                            TAG, "Unknown status when creating private share"
+                        )
+                    }
+                }
+            }
+        )
+    }
+
+    private fun getAppropiatePermissions(shareType: ShareType?): Int {
+        // check if the Share is FERERATED
+        val isFederated = ShareType.FEDERATED == shareType
+
+        when {
+            file.isSharedWithMe -> return RemoteShare.READ_PERMISSION_FLAG    // minimum permissions
+            isFederated -> {
+                val serverVersion = com.owncloud.android.authentication.AccountUtils.getServerVersion(account)
+                return if (serverVersion != null && serverVersion.isNotReshareableFederatedSupported) {
+                    if (file.isFolder)
+                        RemoteShare.FEDERATED_PERMISSIONS_FOR_FOLDER_AFTER_OC9
+                    else
+                        RemoteShare.FEDERATED_PERMISSIONS_FOR_FILE_AFTER_OC9
+                } else {
+                    if (file.isFolder)
+                        RemoteShare.FEDERATED_PERMISSIONS_FOR_FOLDER_UP_TO_OC9
+                    else
+                        RemoteShare.FEDERATED_PERMISSIONS_FOR_FILE_UP_TO_OC9
+                }
+            }
+            else -> return if (file.isFolder)
+                RemoteShare.MAXIMUM_PERMISSIONS_FOR_FOLDER
+            else
+                RemoteShare.MAXIMUM_PERMISSIONS_FOR_FILE
+        }
+    }
+
+    private fun observePrivateShareEdition() {
+        ocShareViewModel.privateShareEditionStatus.observe(
+            this,
+            Observer { uiResult ->
+                when (uiResult?.status) {
+                    Status.ERROR -> {
+                        Snackbar.make(
+                            findViewById(android.R.id.content),
+                            uiResult.errorMessage!!,
+                            Snackbar.LENGTH_SHORT
+                        ).show()
+                    }
+                    Status.LOADING -> {
+                        showLoadingDialog(R.string.common_loading)
+                    }
+                    else -> {
```

Comment:
nothing to do with success case??

Reply:
No, I'm observing the database shares from `observeShares()` function of ShareFileFragment so if everything was fine, they are automatically updated from there.